enemy-test/src/enemy_monster.c

```diff
@@ -14,49 +14,58 @@ void init_enemy(ENEMY* unit, UINT8 enemy_sprite_l, UINT8 enemy_sprite_r, UINT8 s
 	unit->damage = damage;
 	unit->frames_between_attacks = frames_between_attacks;
 	unit->frames_until_next_attack = frames_between_attacks; // There is one full cycle before the enemy starts behaving normally. Might want to be able to configure that.
+	unit->dying_animation_state = 0; // Start off alive.
 }
 
 // Display enemy unit on-screen at specified x and y coordinates
-void display_enemy(ENEMY* unit, UINT8 xpos, UINT8 ypos) {
-	// Store current position
-	unit->xpos = xpos;
-	unit->ypos = ypos;
-	
+void display_enemy(ENEMY* unit, UINT8 xpos, UINT8 ypos) {	
 	// Initialize left sprite
 	set_sprite_tile(unit->sprite_id, unit->enemy_sprite_l);
-	move_sprite(unit->sprite_id, xpos - 8, ypos - 16);
 	
 	// Initialize right sprite
 	set_sprite_tile(unit->sprite_id + 1, unit->enemy_sprite_r);
-	move_sprite(unit->sprite_id + 1, xpos, ypos - 16);
+	
+	// Place the sprite
+	move_enemy(unit, xpos, ypos);
 }
 
-// Play death sequence (blinking), then make the enemy disappear
-void enemy_death(ENEMY* unit) {
-	// Blinking animation - disabled
-	/*
-	UINT8 blinking_cycles = 0;
-	UINT8 cycle_state = 1;
+void move_enemy(ENEMY* unit, UINT8 xpos, UINT8 ypos) {
+	// Moves enemy unit to (x,y)
+	move_sprite(unit->sprite_id, xpos + 8, ypos + 16);
+	move_sprite(unit->sprite_id + 1, xpos + 16, ypos + 16);
 	
-	UINT8 former_ypos = unit->ypos; // Retain former y pos for the blinking animation
-	
-	while (blinking_cycles < 10) {	
-		if (cycle_state) { // Disappears
-			move_sprite(unit->sprite_id, unit->xpos, 0);
-			move_sprite(unit->sprite_id+1, unit->xpos+8, 0);
-			cycle_state = 0;
-		} else {
-			move_sprite(unit->sprite_id, unit->xpos, former_ypos);
-			move_sprite(unit->sprite_id+1, unit->xpos+8, former_ypos);
-			cycle_state = 1;
-			blinking_cycles++;
-		}
-	}*/
+	// Store current position unless it's out of range
+	if (xpos < 170 && ypos < 170)
+	{
+		unit->xpos = xpos;
+		unit->ypos = ypos;
+	}
+}

```

Comment:
FIXME: Won't work with big rooms. Not being visible doesn't mean not existing!

Reply:
(You can safely assume that enemies further than (220, 200) will never be visible however)